gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceRecordIterator.java

```diff
@@ -99,7 +103,40 @@ private void fulfillCurrentRecord() {
     }
   }
 
-  private void resetCurrentRecordStatus() {
+  private InputStreamCSVReader reopenStreamWithRetry(ResultStruct resultStruct, int seekLineNumber, int retryNumber, int count) {
```

Comment:
Can you refactor the logic in `SalesforceExtractor.reinitializeBufferedReader` and use it here? That has an additional validation to make sure that after skipping we arrived at the expected location based on the last record read before the failure.

Reply:
Thanks! I did a refactor to make both pk-chunking and normal bulk batch using Iterator.